docs/installing-open-mpi/configure-cli-options/installation.rst

```diff
@@ -32,48 +30,100 @@ be used with ``configure``:
   Be sure to read the description of ``--without-memory-manager``,
   below; it may have some effect on ``--enable-static``.
 
-* ``--disable-wrapper-rpath``:
-  By default, the wrapper compilers (e.g., ``mpicc``) will enable
-  "rpath" support in generated executables on systems that support it.
-  That is, they will include a file reference to the location of Open
-  MPI's libraries in the application executable itself.  This means
-  that the user does not have to set ``LD_LIBRARY_PATH`` to find Open
-  MPI's libraries (e.g., if they are installed in a location that the
-  run-time linker does not search by default).
-
-  On systems that utilize the GNU ``ld`` linker, recent enough versions
-  will actually utilize "runpath" functionality, not "rpath".  There
-  is an important difference between the two:
-
-  #. "rpath": the location of the Open MPI libraries is hard-coded into
-     the MPI/OpenSHMEM application and cannot be overridden at
-     run-time.
-  #. "runpath": the location of the Open MPI libraries is hard-coded into
-     the MPI/OpenSHMEM application, but can be overridden at run-time
-     by setting the ``LD_LIBRARY_PATH`` environment variable.
+* ``--disable-wrapper-runpath`` / ``--disable-wrapper-rpath``: By
+  default, the wrapper compilers (e.g., ``mpicc``) will enable
+  "runpath" or "rpath" support in generated executables on systems
+  that support it.  That is, they will include a filesystem path
+  reference to the location of Open MPI's libraries in the application
+  executable itself.  This means that the user does not have to set
+  ``LD_LIBRARY_PATH`` to find Open MPI's libraries, which can be
+  helpful if they are installed in a location that the run-time linker
+  does not search by default.
+
+  .. note:: By default, the wrapper compilers prefer "runpath"
+            behavior over "rpath" behavior.
+
+            * Using ``--disable-wrapper-runpath`` alters this
+              preference: "runpath" behavior will never be used.
+              Instead, the wrapper compilers will use "rpath" behavior
```

Comment:
This is incorrect. `--disable-wrapper-runpath` will drop `-Wl,--enable-new-dtags` and leave it up to the compiler defaults (which is runpath for ages already in GCC).

https://github.com/open-mpi/ompi/issues/10560

Reply:
Gah; you're right.  Thanks for catching this.  Reuti also offered some suggestions in https://www.mail-archive.com/users@lists.open-mpi.org/msg34968.html; I'll make a follow-on PR.

Reply:
Could the follow-up PR be a fix that makes `--enable-wrapper-runpath` set `-Wl,--enable-new-dtags` and `--enable-wrapper-rpath` -> `-Wl,--disable-new-dtags`? To me it's a bug in openmpi that explicitly asking to enable rpath and disable runpath leaves you with linker defaults, which is runpath in almost all cases

Reply:
That's a fair point.

I made the follow-on PR #10652 -- please have a look.

I did *not* change `configure` or the wrappers behavior yet -- #10652 is about documenting the current behavior.  We should follow up on #10560 to decide on exactly how to change the behavior, and then have a new PR to both change the behavior and update the docs to reflect the new behavior.